packages/babel-plugin-transform-typescript/src/index.ts

```diff
@@ -465,12 +465,26 @@ export default declare((api, opts) => {
         transpileEnum(path, t);
       },
 
-      TSImportEqualsDeclaration(path) {
-        throw path.buildCodeFrameError(
-          "`import =` is not supported by @babel/plugin-transform-typescript\n" +
-            "Please consider using " +
-            "`import <moduleName> from '<moduleName>';` alongside " +
-            "Typescript's --allowSyntheticDefaultImports option.",
+      TSImportEqualsDeclaration(path: NodePath<t.TSImportEqualsDeclaration>) {
+        if (t.isTSExternalModuleReference(path.node.moduleReference)) {
+          // import alias = require('foo');
+          throw path.buildCodeFrameError(
+            `\`import ${path.node.id.name} = require('${path.node.moduleReference.expression.value}')\` ` +
+              "is not supported by @babel/plugin-transform-typescript\n" +
+              "Please consider using " +
+              `\`import ${path.node.id.name} from '${path.node.moduleReference.expression.value}';\` alongside ` +
+              "Typescript's --allowSyntheticDefaultImports option.",
+          );
+        }
+
+        // import alias = Namespace;
+        path.replaceWith(
+          t.variableDeclaration("var", [
```

Comment:
should we use `var` directly here(The same as tsc)? or leave an option for user to decide?

Reply:
It's ok to always use var.